### PR TITLE
Integrate fetching of documents into frontend

### PR DESCRIPTION
**Pull request by ganhongyao**
Changes
- Integrate api calls into frontend 
  - When logged in, the user should see all the documents he/she uploaded in the past.
  - Users are able to upload files. When files are uploaded, they will be taken to the document edit view.
  - Users are able to select files from the main page. When a file is selected, they will be taken to the edit view.
- Add loading spinners during API calls

### CODE REVIEW

**Comment by ganhongyao** [comment]
@zognin 

I tagged you as reviewer too since this involves views, so I just wanted to make sure that I'm not doing anything wrong

File: Annotato/Annotato/ViewModels/Sample/SampleData.swift [hunk 1]
```diff
@@ -1,62 +1,46 @@
 import CoreGraphics
 import Foundation
+import AnnotatoSharedLibrary
 
 class SampleData {
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/ViewModels/Sample/SampleData.swift:5]: I think we can safely delete this entire file. If don't fall back on these when some error happens.

File: Annotato/Annotato/Views/Components/Protocols/SpinnerPresentable.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import UIKit
+
+protocol SpinnerPresentable where Self: UIViewController {
+    var spinner: UIActivityIndicatorView { get }
+    func initializeSpinner()
+    func startSpinner()
+    func stopSpinner()
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/Views/Components/Protocols/SpinnerPresentable.swift]:
```suggestion
    var spinner: UIActivityIndicatorView { get }
    
    func initializeSpinner()
    func startSpinner()
    func stopSpinner()
```

File: Annotato/Annotato/Views/Components/Protocols/SpinnerPresentable.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+extension SpinnerPresentable {
+    func initializeSpinner() {
+        spinner.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(spinner)
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/Views/Components/Protocols/SpinnerPresentable.swift:13]: Should the spinner be added last?

File: Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift [hunk 4]
```diff
@@ -11,12 +13,25 @@ class DocumentEditViewController: UIViewController, AlertPresentable {
     }
 
     func initializeSubviews() {
+        initializeSpinner()
+
         initializeToolbar()
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift:18]:
```suggestion
        initializeSpinner()
        initializeToolbar()
        startSpinner()
```

Should we start here?

File: Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift [hunk 5]
```diff
@@ -11,12 +13,25 @@ class DocumentEditViewController: UIViewController, AlertPresentable {
...
-            presentErrorAlert(errorMessage: "No document view model")
-            return
+
+        Task {
+            guard let documentId = documentId else {
+                AnnotatoLogger.info("Could not get current user, sample document will be used",
+                                    context: "DocumentEditViewController::initializeSubviews")
+                documentViewModel = DocumentViewModel(document: SampleData.exampleDocument)
+                initializeDocumentView()
+                return
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift]: I'm pretty confident that our app will work? Should we remove this?

File: Annotato/Annotato/Views/Document/List/DocumentListViewController.swift [hunk 6]
```diff
@@ -31,13 +34,37 @@ class DocumentListViewController: UIViewController {
...
 
     private func initializeDocumentsCollectionView() {
+        Task {
+            guard let userId = AnnotatoAuth().currentUser?.uid else {
+                AnnotatoLogger.info("Could not get current user, sample documents will be used",
+                                    context: "DocumentListViewController::initializeSubviews")
+
+                documents = SampleData.exampleDocumentsInList
+                addDocumentsSubview()
+                return
```
Comment (sivayogasubramanian) [inline Annotato/Annotato/Views/Document/List/DocumentListViewController.swift]: Same as above. But I think its safer to have it? IDK LOL

**Review by sivayogasubramanian — APPROVED** [review]
LGTM! Some minor things. I will approve it now and you can merge whenever.

File: Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift (hunk 4 above)
Comment (ganhongyao) [reply Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift:18]: I think this makes it hard to find the matching start and stop spinner actions, and can be prone to developer error in future?

**Pull request merged**